File: src/Goods.BackOffice/client-app/src/domain/vehicles/vehicleBlank.ts
import { VehicleCategory } from "./enums/vehicleCategory";
import { Vehicle } from "./vehicle";

export interface VehicleBlank {
  id: string | null;
  driverId: string | null;
  name: string | null;
  stateNumber: string | null;
  vehicleCategory: VehicleCategory | null;
  averageSpeed: number | null;
  fuelConsumption: number | null;
}

export namespace VehicleBlank {
  export function getDefault(): VehicleBlank {
    return {
      id: null,
      driverId: null,
      name: null,
      stateNumber: null,
      vehicleCategory: null,
      averageSpeed: null,
      fuelConsumption: null
    };
  }

  export function fromVehicle(vehicle: Vehicle): VehicleBlank {
    return {
      id: vehicle.id,
      driverId: vehicle.driverId,
      name: vehicle.name,
      stateNumber: vehicle.stateNumber,
      vehicleCategory: vehicle.vehicleCategory,
      averageSpeed: vehicle.averageSpeed,
      fuelConsumption: vehicle.fuelConsumption
    }
  }
}

File: src/Goods.Domain/Vehicles/VehicleBlank.cs
using Goods.Domain.Vehicles.Enums;

namespace Goods.Domain.Vehicles;

public class VehicleBlank
{
    public Guid? Id { get; set; }
    public Guid DriverId { get; set; }
    public String? Name { get; set; }
    public String? StateNumber { get; set; }
    public VehicleCategory? VehicleCategory { get; set; }
    public Decimal? AverageSpeed { get; set; }
    public Decimal? FuelConsumption { get; set; }
    public DateTime? CreatedDatetimeUTC { get; set; }
    public DateTime ModifiedDatetimeUTC { get; set; }
    public Boolean IsRemoved { get; set; }
}
